mediabridge/main.py

```diff
@@ -1,31 +1,49 @@
 import logging
+import os
+from datetime import datetime
 
 import typer as typer
 from tqdm.contrib.logging import logging_redirect_tqdm
 
 from mediabridge.data_processing import wiki_to_netflix
 
 
-def main(verbose: bool = typer.Option(False, "--verbose", "-v"), test: bool = False):
-    if test:
+def main(
+    verbose: bool = typer.Option(False, "--verbose", "-v"),
+    log: bool = False,
+    test: bool = False,
+):
+    # create output directory if it doesn't exist
+    if not os.path.exists("./out"):
```

Comment:
This will create the `out` directory in whatever directory the script was run from. While we expect that this will usually be the MediaBridge root directory, there are no guarantees.

Reply:
I'm aware but not sure how we'd work around this. The 'mediabridge' module imports fail if you aren't using `pipenv run` in the root project directory, so it shouldn't be possible to work around this, but I can't come up with an explicit way to always ensure the main script is run from the root.

Reply:
We could check if the cwd path ends with MediaBridge but that seems a little silly. And yeah, make does exist.

Reply:
Think about it the other way, how do we ensure that the directory is created in the same place, regardless of where the script is run from.

We can use the `__file__` built-in to get the path to the file we are currently in. If we know that, we know for sure where the `out` directory should be.

In fact, we already do this caclulation [here](https://github.com/noisebridge/MediaBridge/blob/main/mediabridge/data_processing/wiki_to_netflix.py#L29). You should just `mkdir` that variable.

Reply:
Oh sorry, I'm confused. This is `mediabridge/main.py` not `wiki_to_netflix.py`, where that variable is. In #21, @smaysenhalder is adding a constants file that has the directory paths in it. That would clearly be very useful here. Maybe put a TODO to wait until those constants are ready and skip creating the output directory for now? Or just recalculate the output directory like `wiki_to_netflix.py` does and put a TODO to remove the duplication once it's in the constants file?

Reply:
I don't know how I didn't already know about the __file__ special symbol. I just used it in the latest commit to localize our path definitions in a py file not knowing about the work in #21 to ensure we won't make accidental string manipulation errors.